boa_engine/src/bytecompiler/expression/object_literal.rs

```diff
@@ -57,12 +58,20 @@ impl ByteCompiler<'_, '_> {
                             self.object_method(expr.into());
                             self.emit_opcode(Opcode::SetFunctionName);
                             self.emit_u8(1);
+                            self.emit_opcode(Opcode::RotateLeft);
+                            self.emit_u8(3);
+                            self.emit_opcode(Opcode::Swap);
+                            self.emit_opcode(Opcode::SetHomeObject);
+                            self.emit_opcode(Opcode::Swap);
+                            self.emit_opcode(Opcode::RotateRight);
+                            self.emit_u8(3);
```

Comment:
```suggestion
                            // stack: object, object
                            // stack: object, object, name
                            // stack: object, object, ToPropertyKey(name)
                            // stack: object, object, ToPropertyKey(name), ToPropertyKey(name) 
                            self.object_method(expr.into());
                            
                            // stack: object, object, ToPropertyKey(name), ToPropertyKey(name), method
                            self.emit_opcode(Opcode::SetFunctionName);
                            self.emit_u8(1);
                            
                            // stack: object, object, ToPropertyKey(name), method
                            self.emit_opcode(Opcode::RotateLeft);
                            self.emit_u8(3);
                            
                            // stack: object, ToPropertyKey(name), method, object
                            self.emit_opcode(Opcode::Swap);
                            
                            // stack: object, ToPropertyKey(name), object, method
                            self.emit_opcode(Opcode::SetHomeObject);
                            
                            // stack: object, ToPropertyKey(name), object, method
                            self.emit_opcode(Opcode::Swap);
                            
                            // stack: object, ToPropertyKey(name), method, object
                            self.emit_opcode(Opcode::RotateRight);
                            self.emit_u8(3);
                            
                            // stack: object, object, ToPropertyKey(name), method
```

I think we should introduce some documentation into the bytecompiler of the current stack state, something like this, because all the stack juggling is confusing sometimes.

I think it would help with debugging as well as adding new features